src/shared/awsContext.ts

```diff
@@ -55,6 +59,40 @@ export class DefaultAwsContext implements AwsContext {
         const persistedRegions = context.globalState.get<string[]>(regionSettingKey)
         this.explorerRegions = persistedRegions || []
     }
+    /**
+    
+        Checks in order of precedence
+       1. explorerRegion[0] if only one region exists.
+       2. the node passed to the current command
+       3. last-expanded AWS Explorer service / etc
+       4. last selection region in RegionProvider of Wizard. 
+       5. final fallback is the credentials default region
+
```

Comment:
Outdated documentation, the precedence is now wrong. Writing the precedence is probably too specific as callers don't really care how the heuristic is computed with the exception of the `node` parameter. And if they did care, the tests should describe the behavior.

I'd try to limit docstrings to contain high-level ideas with as few implementation details as possible.